report.md

```diff
@@ -4,123 +4,157 @@
-de-identifiable and identifiable data are processed locally and on the
-UCL Data Safe haven (IDHS) respectively.  In order to create the pipeline on IDHS, we
-requested and configured a Linux virtual environment. Due to the limited capacity of
-IDHS, it is still useful to keep a local de-identifiable copy for development
-purposes. The pipeline is designed to be portable on multiple platforms. It can be
-executed on both local environments and IDHS with de-identifiable and 
-identifiable data. 
+contains demographic, drugs, laboratory, nursing and physiology data. Both
+de-identified and identifiable data are processed, locally and on the
+UCL Data Safe Haven (IDHS) respectively.  In order to create the pipeline on IDHS, UCL's
+standards-compliant solution for working with restricted and confidential data, we
+requested and configured an appopriate Linux virtual environment. 
+
+Due to the limited computational capacity of
+IDHS, it is still useful to keep a local de-identified copy of the data for development
+purposes. The pipeline is designed to be portable to multiple platforms, and will additionally be deployed
+to UCH servers in due course.
 
 ## XML parser
 
 The `XML` parser in R combines and restructures the `XML` files into a newly defined R
 data structure `ccRecord`, which significantly improves the clarity of the
-data by organising it under tables and removing the redundancy of the `XML`
+data by organising it into tables and removing significant redundancy in the `XML`
 files. `ccRecord` is designed as a flexible, simplified, and queryable data
-structure for critical care measurements. Data in `ccRecord` format is eventually
+structure for critical care measurements. 
+
+Data in `ccRecord` format is eventually
 stored in a RData file which is about 500 times smaller than the original `XML`
-files.  In addition the data provenance is recorded by each episode, henceforth
+files.  
+
+Data provenance is recorded for each episode, thus
 we are able to tell from which file each episode comes and when
-it has been parsed. The selected data fields can be exported into a `CSV` file for
+it has been parsed. Selected data fields can be exported into a `CSV` file for
 Excel users or to be used with any other analysis tool or programming language.
 
-It is worth mentioning that a `C++` equivalent `XML` parser, which has a much better
-performance comparing to the currently used in R, has been developed but not deployed yet due
-to the time constraints of the project. The incorporation of the `C++` parser into the
-pipeline will boost the speed in which all the files are converted.
+We have on going work to develop an equivalent parser in `C++`. The prototype has a much better
+performance comparing to the `R` version, but has not yet been deployed into the current pipeline. 
+The incorporation of the `C++` parser into the
+pipeline will reduce from Xs to Ys the time to parse all the files.
```

Comment:
Probably we need to get these numbers, right? @sinanshi let me know how I can get these numbers.